lib/netease.rb

```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+require "excon"
+
+class Netease
+  MAXIMUM_CONTENT_LENGTH = 9999
+
+  class Error < StandardError
+  end
+
+  class RequestArgs
+    def initialize(target, munger = nil)
+      @target = target
+      @munger = munger
+    end
+
+    def for_check
+      send("for_#{target_name}_check")
+    end
+
+    def for_feedback
+      { feedback: { taskId: @target.custom_fields["NETEASE_TASK_ID"] } }
+    end
+
+    private
+
+    def for_post_check
+      { dataId: "post-#{@target.id}", content: post_content&.strip[0..MAXIMUM_CONTENT_LENGTH] }
+    end
+
+    def for_user_check
+      bio = @target.user_profile&.bio_raw
+
+      { dataId: "user-#{@target.id}", content: bio&.strip[0..MAXIMUM_CONTENT_LENGTH] }
+    end
+
+    def target_name
+      @target.class.to_s.downcase
+    end
+
+    def post_content
+      return if !@target.is_a?(Post)
+      return @target.raw if !@target.is_first_post?
+
+      topic = @target.topic || Topic.with_deleted.find_by(id: @target.topic_id)
+      "#{topic && topic.title}\n\n#{@target.raw}"
+    end
+  end
+
+  class Client
+    PLUGIN_NAME = "discourse-akismet"
+    CHECK_VERSION = "v5.2"
+    FEEDBACK_VERSION = "v2"
+    FEEDBACK_PATH = "v2/text/feedback"
+    CHECK_PATH = "v5/text/check"
+    API_BASE_URL = "http://as.dun.163.com"
+    HASH_ALGORITHM = "MD5"
+    NONCE_UPPER_BOUND = 100_000 * 100_000 * 10
```

Comment:
What is happening here?

Reply:
The API requires a nonce that can be up to 11 digits long.  The` NONCE_UPPER_BOUND` is  the argument to the  `SecureRandom.random_number`.  I opted for `100_000 * 100_000 * 10` instead of `10_000_000_0000` because I felt it was more readable.